packages/main/src/StepInput.ts

```diff
@@ -441,12 +470,17 @@ class StepInput extends UI5Element implements IFormElement {
 	}
 
 	_updateValueState() {
-		const valid = !((this.min !== undefined && this.value < this.min) || (this.max !== undefined && this.value > this.max));
+		const isWithinRange = (this.min === undefined || Number(this.input.value) >= this.min)
+							  && (this.max === undefined || Number(this.input.value) <= this.max);
+		const isPrecisionValid = this._isPrecisionValid();
 		const previousValueState = this.valueState;
 
-		this.valueState = valid ? ValueState.None : ValueState.Error;
+		this.valueState = isWithinRange && isPrecisionValid ? ValueState.None : ValueState.Error;
```

Comment:
isWithinRange && isPrecisionValid is used in two different places, we can have a variable for it. isValid = isWithinRange && isPrecisionValid;